movingpandas/tests/test_cpa.py

```diff
@@ -0,0 +1,467 @@
+
+    geometry_a = [shapely.Point(*point) for point in points_a]
+    df_a = pd.DataFrame(data=dict(t=t_a))
+    gdf_a = gpd.GeoDataFrame(df_a, geometry=geometry_a, crs=CRS)
+    traj_a = mpd.Trajectory(gdf_a, traj_id="traj_a", t="t", crs=CRS)
+
+    geometry_b = [shapely.Point(*point) for point in points_b]
+    df_b = pd.DataFrame(data=dict(t=t_b))
+    gdf_b = gpd.GeoDataFrame(df_b, geometry=geometry_b, crs=CRS)
+    traj_b = mpd.Trajectory(gdf_b, traj_id="traj_b", t="t", crs=CRS)
+
+    cpa = CPACalculator(traj_a, traj_b)
+    result = cpa.min()
+
+    # ASSERT_DOUBLE_EQUAL(m, 3.0);
+    # ASSERT_DOUBLE_EQUAL(dist, 1.0);
+    assert result.t_at.timestamp() == 3, "t should equal 3"
+    assert result.dist == 1, "distance should equal 1"
+    assert result.status == "approaching", "status should be approaching"
+
+
+def test_miliseconds():
+    """Test related to subsecond timestamps
+    https://trac.osgeo.org/postgis/ticket/3136"""
+
+    points_a = [(0, 0), (2, 0)]
+    t_a = [1432291464, 1432291536]
+    # convert to datetime
+    t_a = [datetime.datetime.fromtimestamp(t, tz=UTC).replace(tzinfo=None) for t in t_a]
+    geometry_a = [shapely.Point(*point) for point in points_a]
+    df_a = pd.DataFrame(data=dict(t=t_a))
+    gdf_a = gpd.GeoDataFrame(df_a, geometry=geometry_a, crs=CRS)
+    traj_a = mpd.Trajectory(gdf_a, traj_id="traj_a", t="t", crs=CRS)
+
+    points_b = [(0, 0), (1, 0), (2, 0)]
+    t_b = [1432291464, 1432291466.25, 1432291500]
+    # convert to datetime
+    t_b = [datetime.datetime.fromtimestamp(t, tz=UTC).replace(tzinfo=None) for t in t_b]
+    geometry_b = [shapely.Point(*point) for point in points_b]
+    df_b = pd.DataFrame(data=dict(t=t_b))
+    gdf_b = gpd.GeoDataFrame(df_b, geometry=geometry_b, crs=CRS)
+    traj_b = mpd.Trajectory(gdf_b, traj_id="traj_b", t="t", crs=CRS)
+
+    cpa = CPACalculator(traj_a, traj_b)
+    result = cpa.min()
+    # ASSERT_DOUBLE_EQUAL(m, 1432291464.0);
+    # ASSERT_DOUBLE_EQUAL(dist, 0.0);
+
+    np.testing.assert_almost_equal(
+        result.t_at.timestamp(), 1432291464.0, err_msg="t should equal 1432291464.0"
+    )
+    assert result.dist == 0, "distance should equal 0"
+    assert result.status == "approaching", "status should be approaching"
+
+
+def test_single_point():
+    """Tracks share a single point in time"""
+
+    traj_a = create_traj((0, 0), (1, 0), 0, 2)
+    traj_b = create_traj((0, 0), (1, 0), 2, 3)
+
+    cpa = CPACalculator(traj_a, traj_b)
+    result = cpa.min()
+
+    # ASSERT_DOUBLE_EQUAL(m, 2.0);
+    # ASSERT_DOUBLE_EQUAL(dist, 1.0);
+    #
+    assert result.t_at.timestamp() == 2, "t should equal 2"
+    assert result.dist == 1, "distance should equal 1"
+    assert result.status == "touching", "status should be touching"
+
+
+def test_ais_dk(data_dir):
+    """Two ships in the AIS.dk test set."""
+
+    # data src: http://web.ais.dk/aisdata/
+    mmsi_a = 265550210
+    mmsi_b = 265410000
+    ais_gdf = gpd.read_file(data_dir / "ais-utm32N.json")
```

Comment:
What does the test on this large file add? Compared to the simple tests, it is hard top verify if the results are correct.

Reply:
I reduced the file to the timestamps where the two objects are close together. I consider this a regression test. 
If you think it doesn't contribute enough, I'm happy to remove it.